src/ToDoList.js

```diff
@@ -11,90 +11,163 @@ let taskData = [
     id: 1,
     taskName: "sample task",
     taskDescription: "sample description",
+    isComplete: false,
+  },
+  {
+    id: 2,
+    taskName: "sample task",
+    taskDescription: "sample description",
+    isComplete: true, 
   },
 ];
-
 const ToDoList = () => {
   const [isOpen, setOpen] = useState(false);
-  const [taskCard, setTaskCard] = useState(taskData);
+  const [taskList, setTaskCard] = useState(taskData);
+ 
+  const updateStatusHandler = (id) => {
+    console.log(id);
+    var newTaskData = [];
+    for (var i = 0; i < taskList.length; i++) {
+      if (taskList[i].id == id) {
+        taskList[i].isComplete = !taskList[i].isComplete;
+        newTaskData.push(taskList[i]);
+      } else {
+        newTaskData.push(taskList[i]);
+      }
+    }
+    setTaskCard(newTaskData);
+  };
+
   const handleAddTask = () => {
     setOpen(true);
   };
+
+  const editHandler = (updatedTaskName,updatedTaskDescription,id) => {
+  console.log('edit2 call',id)
+   var newEditedTaskData = [];
+   for (var i = 0; i < taskList.length; i++) {
+     if (taskList[i].id == id) {
+      taskList[i].key = taskList[i].key
+      taskList[i].id = taskList[i].id
+      taskList[i].taskName = updatedTaskName;
+      taskList[i].taskDescription = updatedTaskDescription;
+      taskList[i].isComplete =  taskList[i].isComplete
+      newEditedTaskData.push(taskList[i]);
+     } else {
+      newEditedTaskData.push(taskList[i]);
+    }
+  }
+  console.log(newEditedTaskData)
+  setTaskCard(newEditedTaskData);
+  }
```

Comment:
```suggestion
  const editHandler = (updatedTaskName,updatedTaskDescription,id) => {
   var newEditedTaskData = [];
   for (var i = 0; i < taskList.length; i++) {
     if (taskList[i].id == id) {
      taskList[i].key = taskList[i].key
      taskList[i].id = taskList[i].id
      taskList[i].taskName = updatedTaskName;
      taskList[i].taskDescription = updatedTaskDescription;
      taskList[i].isComplete =  taskList[i].isComplete
     }
      newEditedTaskData.push(taskList[i]);
  }
  setTaskCard(newEditedTaskData);
  }
```

1. Remove Console log.
2.  If `newEditedTaskData.push(taskList[i]);` is required both in `if` & `else` then move it to outside the if-else scope.